kube-core/src/object.rs

```diff
@@ -102,6 +102,46 @@ impl<'a, T: Clone> IntoIterator for &'a mut ObjectList<T> {
     }
 }
 
+/// A trait to access the `spec` of a Kubernetes resource.
+///
+/// Some built-in Kubernetes resources and all custom resources do have a `spec` field.
+/// This trait can be used to access this field.
+///
+/// This trait is automatically implemented by the kube-derive macro and is _not_ currently
+/// implemented for the Kubernetes API objects from `k8s_openapi`.
+///
+/// Note: Not all Kubernetes resources have a spec (e.g. `ConfigMap`, `Secret`, ...).
+pub trait HasSpec {
+    /// The type of the `spec` of this resource
+    type Spec;
+
+    /// Returns a reference to the `spec` object
+    fn spec(&self) -> &Self::Spec;
+
+    /// Returns a mutable reference to the `spec` object
+    fn spec_mut(&mut self) -> &mut Self::Spec;
+}
+
+/// A trait to access the `status` of a Kubernetes resource.
+///
+/// Some built-in Kubernetes resources and custom resources do have a `status` field.
+/// This trait can be used to access this field.
+///
+/// This trait is automatically implemented by the kube-derive macro and is _not_ currently
+/// implemented for the Kubernetes API objects from `k8s_openapi`.
+///
+/// Note: Not all Kubernetes resources have a status (e.g. `ConfigMap`, `Secret`, ...).
+pub trait HasStatus {
+    /// The type of the `status` object
+    type Status;
+
+    /// Returns an optional reference to the `status` of the object
```

Comment:
```suggestion
    /// Returns an optional reference to the `status` object
```

I kind of prefer the other wording, but we should at least be consistent

Reply:
What do you mean with _other_ wording? I don't have an opinion so if you do I'm happy to adjust/expand.

Reply:
I think he's referring to his suggested change's wording which you can commit there.

Reply:
I was referring to the`foo of the object` wording

Reply:
I've used that wording everywhere now.

Reply:
:+1: